consensus/src/round_manager.rs

```diff
@@ -1222,6 +1224,47 @@ impl RoundManager {
         result
     }
 
+    async fn new_qc_from_order_vote_msg(
+        &mut self,
+        order_vote_msg: &OrderVoteMsg,
+    ) -> anyhow::Result<()> {
+        ensure!(
+            order_vote_msg
+                .order_vote()
+                .ledger_info()
+                .consensus_block_id()
+                == order_vote_msg.quorum_cert().certified_block().id(),
+            "QuorumCert attached to order vote doesn't match"
+        );
+
+        if let NeedFetchResult::QCAlreadyExist = self
+            .block_store
+            .need_fetch_for_quorum_cert(order_vote_msg.quorum_cert())
+        {
+            return Ok(());
+        }
+
+        let start = Instant::now();
+        order_vote_msg
+            .quorum_cert()
+            .verify(&self.epoch_state().verifier)
```

Comment:
Hmm.. I have concerned about moving this verification from dedicated threadpool to single-threaded round manager - may be its okay because we are verifying only one message per round. Do you know how much time it takes?

Reply:
The verification in round manager won't happen frequently. It will happen only if the node doesn't already have the quorum cert of the block by the time it gets the order vote. And it happens only once per block. So at max, 5 verifications per second.

Reply:
Do you know the verification time per QC?

Reply:
Less than a millisecond.

Reply:
Do you have the grafana chart?

Reply:
This was order vote verification time on the main branch. It took about 2.5ms: https://aptoslabs.grafana.net/d/consensus/consensus?from=1722008561000&to=1722009337000&var-Datasource=fHo-R604z&var-BigQuery=axNEitxVz&var-metrics_source=All&var-chain_name=forge-0&var-cluster=All&var-namespace=forge-e2e-pr-14107&var-kubernetes_pod_name=All&orgId=1&editPanel=252

This is the order vote verification time on this branch. It took about 1.3ms. https://aptoslabs.grafana.net/d/consensus/consensus?from=1722011303000&to=1722012155000&var-Datasource=fHo-R604z&var-BigQuery=axNEitxVz&var-metrics_source=All&var-chain_name=forge-0&var-cluster=All&var-namespace=forge-e2e-pr-14129&var-kubernetes_pod_name=All&orgId=1&viewPanel=252

The difference being the verification time of the QC. So, the QC verification time should be about 1.2ms.